comet-java-client/src/test/java/ml/comet/experiment/impl/OnlineExperimentTest.java

```diff
@@ -58,6 +65,63 @@
     private static final String LOGGED_ERROR_LINE = "This error should also get to Comet ML.";
     private static final String NON_LOGGED_LINE = "This should not end up in Comet ML.";
 
+    private static Path root;
+    private static Path emptyFile;
+
+    @BeforeAll
+    static void setup() throws IOException {
+        // create temporary directory tree
+        root = Files.createTempDirectory("testFileUtils");
+        PathUtils.copyFileToDirectory(
+                Objects.requireNonNull(TestUtils.getFile(SOME_TEXT_FILE_NAME)).toPath(), root);
+        PathUtils.copyFileToDirectory(
+                Objects.requireNonNull(TestUtils.getFile(ANOTHER_TEXT_FILE_NAME)).toPath(), root);
+        emptyFile = Files.createTempFile(root, "c_file", ".txt");
+
+        Path subDir = Files.createTempDirectory(root, "subDir");
+        PathUtils.copyFileToDirectory(
+                Objects.requireNonNull(TestUtils.getFile(IMAGE_FILE_NAME)).toPath(), subDir);
+        PathUtils.copyFileToDirectory(
+                Objects.requireNonNull(TestUtils.getFile(CODE_FILE_NAME)).toPath(), subDir);
+    }
+
+    @AfterAll
+    static void tearDown() throws IOException {
+        PathUtils.delete(root);
+        assertFalse(Files.exists(root), "Directory still exists");
+    }
+
+    @Test
+    public void testLogAndGetAssetsFolder() {
+        OnlineExperimentImpl experiment = (OnlineExperimentImpl) createOnlineExperiment();
+
+        // Make sure experiment has no assets
+        //
+        assertTrue(experiment.getAssetList(ASSET_TYPE_ALL).isEmpty());
+
+        // Log assets folder nd wait for completion
+        //
+        ExperimentContext context = new ExperimentContext(123, 1042, "train");
+        OnCompleteAction onComplete = new OnCompleteAction();
+        experiment.logAssetFolder(root.toFile(), false, true, context, onComplete);
+
+        awaitForCondition(onComplete, "log assets' folder timeout", 60);
+
+        // wait for assets become available and validate results
+        //
+        awaitForCondition(() -> experiment.getAssetList(ASSET_TYPE_ALL).size() == 5, "Assets was uploaded");
```

Comment:
test is valid. its better to get the expected (in this case `5` ) dynamic and not hard code this. for the reader i dont have all the information to understand the test by reading it and will require to go to the root.toFile() and see that we actually have 5 files in it. better to scan list and get size of list on assertion. this way the reader can understand everything in the test

Reply:
Fixed